test/external-modules/latency-simulating-directory/src/internalClusterTest/java/org/elasticsearch/test/simulatedlatencyrepo/LatencySimulatingBlobStoreRepositoryTests.java

```diff
@@ -0,0 +1,139 @@
+import org.elasticsearch.xpack.core.searchablesnapshots.MountSearchableSnapshotRequest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.LongAdder;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
+public class LatencySimulatingBlobStoreRepositoryTests extends AbstractSnapshotIntegTestCase {
+
+    public static final String REPO_TYPE = "countingFs";
+
+    public static final LongAdder COUNTS = new LongAdder();
+
+    public static class TestPlugin extends Plugin implements RepositoryPlugin {
+
+        @Override
+        public Map<String, Repository.Factory> getRepositories(
+            Environment env,
+            NamedXContentRegistry namedXContentRegistry,
+            ClusterService clusterService,
+            BigArrays bigArrays,
+            RecoverySettings recoverySettings
+        ) {
+            return Map.of(
+                REPO_TYPE,
+                metadata -> new LatencySimulatingBlobStoreRepository(
+                    metadata,
+                    env,
+                    namedXContentRegistry,
+                    clusterService,
+                    bigArrays,
+                    recoverySettings,
+                    COUNTS::increment
+                )
+            );
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(TestPlugin.class);
+        plugins.add(LocalStateSearchableSnapshots.class);
+        return plugins;
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        Settings.Builder builder = Settings.builder();
+        builder.put(otherSettings);
+        builder.put(LicenseSettings.SELF_GENERATED_LICENSE_TYPE.getKey(), "trial");
+        return builder.build();
+    }
+
+    public void testRetrieveSnapshots() throws Exception {
+        final Client client = client();
+        final String repositoryName = "test-repo";
+
+        logger.info("-->  creating repository");
+        createRepository(repositoryName, REPO_TYPE, Settings.builder().put("location", randomRepoPath()));
+
+        logger.info("--> creating an index and indexing documents");
+        final String indexName = "test-idx";
+        createIndex(indexName);
+        ensureGreen();
+        int numDocs = randomIntBetween(10, 20);
+        for (int i = 0; i < numDocs; i++) {
+            String id = Integer.toString(i);
+            client().prepareIndex(indexName).setId(id).setSource("text", "sometext").get();
+        }
+        indicesAdmin().prepareFlush(indexName).get();
+
+        logger.info("--> create snapshot");
+        CreateSnapshotResponse createSnapshotResponse = client.admin()
```

Comment:
NIT: You can just use ` createSnapshot(fsRepoName, "test-snap-1", List.of(indexName));` here now :), even does the login for you.